Improving documentation for `QuantumNetwork` class + bug fix

Canoming opened the pull request:
Checklist:
- [x] Reviewers confirm new code works as expected.
- [x] Tests are passing.
- [x] Coverage does not decrease.
- [x] Documentation is updated.

renatomello commented on src/qibo/quantum_info/quantum_networks.py:
```diff
@@ -42,7 +58,7 @@ def __init__(
         self.system_output = system_output
         self._pure = pure
-        self._backend = backend
+        self._backend = backend if backend is not None else GlobalBackend()
```
this line does not need to be changed because the backend check is already done inside `_set_tensor_and_parameters`

AlejandroSopena commented on src/qibo/quantum_info/quantum_networks.py:
```diff
@@ -426,6 +440,11 @@ def __add__(self, second_network):
 
+        If the quantum network is pure and ``number > 0.0``, the method returns a pure quantum
+        network with its Choi operator multiplied by the square root of ``number``.
+        This is equivelant to multiplying `self.to_full()` by the ``number``.
```
```suggestion
        This is equivalent to multiplying `self.to_full()` by the ``number``.
```

AlejandroSopena commented on src/qibo/quantum_info/quantum_networks.py:
```diff
@@ -465,6 +484,11 @@ def __rmul__(self, number: Union[float, int]):
 
+        If the quantum network is pure and ``number > 0.0``, the method returns a pure quantum
+        network with its Choi operator divided by the square root of ``number``.
+        This is equivelant to dividing `self.to_full()` by the ``number``.
```
```suggestion
        This is equivalent to dividing `self.to_full()` by the ``number``.
```

Canoming commented:
> @Canoming let us know if you plan on adding more things to this PR, otherwise we can merge it

I'm trying to write a small use case of `QuantumNetwork`. But I would like to first check if there is a reason we chose the function name `channel` over `is_channel`: https://github.com/qiboteam/qibo/issues/1189#issuecomment-1927172458 . I can commit the use case after we confirm the function name

renatomello commented:
> > @Canoming let us know if you plan on adding more things to this PR, otherwise we can merge it
> 
> I'm trying to write a small use case of `QuantumNetwork`. But I would like to first check if there is a reason we chose the function name `channel` over `is_channel`: [#1189 (comment)](https://github.com/qiboteam/qibo/issues/1189#issuecomment-1927172458) . I can commit the use case after we confirm the function name

It's already changed

Canoming commented:
I have updated the test and added a tutorial for the `QuantumNetwork`. The tests are not ideal at this stage as I didn't test the output of `@`. However, I need to implement a more complex `link_product` for a better test, which is planned for the next step.

renatomello commented on doc/source/code-examples/tutorials/quantum_networks/README.md:
```diff
@@ -0,0 +1,125 @@
+# Quantum Networks
+
+## The Quantum Network Model
+
+The quantum network model is a mathematical framework that allows us to uniquely describe quantum information processing that involves multiple points in time and space.
+Each distinguished point in time and space is treated as a linear system $\mathcal{H}_i$.
+A quantum network involving $n$ points in time and space is a Hermitian operator $\mathcal{N}$ that acts on the tensor product of the linear systems $\mathcal{H}_0 \otimes \mathcal{H}_1 \otimes \cdots \otimes \mathcal{H}_{n-1}$.
+Each system $\mathcal{H}_i$ is either an input or an output of the network.
+
+A physically implementable quantum network is described by a semi-positive definite operator $\mathcal{N}$ that satisfies the causal constraints.
+
+A simple example is a quantum channel $\Gamma: \mathcal{H}_0 \to \mathcal{H}_1$, where $\mathcal{H}_0$ is the input system and $\mathcal{H}_1$ is the output system.
+The quantum channel is a linear map, such that it maps any input quantum state to an output quantum state, which is a sufficient and necessary condition for the map to be physical.
+A Hermitian operator $J^\Gamma$ acting on $\mathcal{H}_0\otimes \mathcal{H}_1$ is associated with a quantum channel $\Gamma$, if $J^\Gamma$ satisfies the following conditions:
+$$
+J^\Gamma \geq 0, \quad \text{and} \quad \text{Tr}_{\mathcal{H}_1} J^\Gamma = \mathbb{I}_{\mathcal{H}_0}.
+$$
+The first condition is called *complete positivity*, and the second condition is called *trace-preserving*.
+In particular, the second condition ensures that the information of the input system is only accessible through the output system.
+
+In particular, a quantum state $\rho$ may be also considered as a quantum network, where the input system is the trivial system $\mathbb{C}$, and the output system is the quantum system $\mathcal{H}$.
+The constraints on the quantum channels are then equivalent to the constraints on the quantum states:
+$$
+\rho \geq 0, \quad \text{and} \quad \text{Tr} \rho = \mathbb{I}_\mathbb{C} = 1.
+$$
```
Some of the equations here are not rendering
